### ISSUE TITLE
NuclioUI Url change from localhost to real url 

### ISSUE THREAD

**Issue by IgorKurylo** [mlrun/ce#51]
We deploy your chart to eks cluster, we try to deploy function, and deployment is working but, when we go to mlrun-ui and click on the function which deployed this redirect to nuclio-ui  but with localhost address, once i replace the localhost with my nuclio url. is working. how i can replace nuclio ui url. when deploy the chart and is impact to mlrun ui.
The following image is mlrun dashboard, look on the bottom we can look the localhost address.
![image](https://user-images.githubusercontent.com/20112133/211310754-f3e4ae61-a49f-40da-aaea-25aab91b75f5.png)
Thanks.

**Comment by liranbg** [comment]
Hi @IgorKurylo - Thank you for raising this issue, can you please let us know what is the `helm` command you have used to spin up MLRun CE on your cloud?

make sure you provide it with either `--set global.externalHostAddress=<external-ip-address>` or `--set mlrun.nuclio.uiURL=<ingress>` if you have ingresses setup.

**Comment by liranbg** [comment]
tl;dr information about creating functions with clusterIP / ingresses.

You can set MLRun to deploy Nuclio functions with `ClusterIP` instead of `NodePort`.
Nnote that those functions *would not be invokable externally* - but only within the cluster, unless you add _ingresses_ as well.

Setting MLRun to work with `ClusterIP` as Kubernetes service type:

```
--set mlrun.api.extraEnvKeyValue.MLRUN_HTTPDB__NUCLIO__DEFAULT_SERVICE_TYPE=ClusterIP
```

If you want MLRun to create the Nuclio function with an ingress as well, then make sure you add below as well:

```
--set mlrun.api.extraEnvKeyValue.MLRUN_HTTPDB__NUCLIO__ADD_TEMPLATED_INGRESS_HOST_MODE=onClusterIP
```

If you wonder about the ingress host it will generate - then, you might need to tweak a bit the `values.yaml` and extend nuclio's platform configuration

```yaml
...
nuclio
  platform:
    ...
    kube:
      defaultHTTPIngressHostTemplate: "{{ .ResourceName }}-{{ .ProjectName }}.{{ .Namespace }}.your-domain.com"
      defaultServiceType: clusterIP
```

If you need ingresses to be created with pre-defined annotations:

add

```yaml
defaultHTTPIngressAnnotations:
  key: value
```
under `nuclio.platform.kube` as well (see above.)

**Comment by IgorKurylo** [comment]
OK, but the ALB Ingress controller works only when the service is on NodePort and not in ClusterIP. so if I change the value to cluster IP I can't set up ingress to function. What is the correct way to deploy functions and expose them via an ingress controller in your ecosystem, I would like to deploy functions and bind them to one ingress which creates dynamic rules.
You can please explain what the best process is for our use cace?

**Comment by liranbg** [comment]
Well, depend on your case and your limitations. if you need to adhere your cloud conventions, then, to create NodePorts:
```
--set mlrun.api.extraEnvKeyValue.MLRUN_HTTPDB__NUCLIO__DEFAULT_SERVICE_TYPE=NodePort
```
and 

```
--set mlrun.api.extraEnvKeyValue.MLRUN_HTTPDB__NUCLIO__ADD_TEMPLATED_INGRESS_HOST_MODE=always
```

same goes for Nuclio's platform configuration
```
...
nuclio
  platform:
    ...
    kube:
      defaultHTTPIngressHostTemplate: "{{ .ResourceName }}-{{ .ProjectName }}.{{ .Namespace }}.your-domain.com"
      defaultServiceType: NodePort
```

That means, your function would be deployed with NodePort by default and will ensure an ingress would be created as well.

**Comment by IgorKurylo** [comment]
OK thanks, I will try it.

**Comment by IgorKurylo** [comment]
@liranbg any suggestion ?

**Comment by liranbg** [comment]
under nuclio's section

**Comment by IgorKurylo** [comment]
Ok, so i need to pull the values.yaml and add 
`kube:
      defaultHTTPIngressHostTemplate: "{{ .ResourceName }}-{{ .ProjectName }}.{{ .Namespace }}.your-domain.com"
      defaultServiceType: NodePort`
or I can put this with --set in helm chart.

**Comment by liranbg** [comment]
hmm you can `kubectl -n <namespace> edit cm nuclio-platform-config` and add it there by yourself after, or need to provide it within a values file. it is a bit tricky or perhps Id say advanced usage to define it but all lego pieces are there.
note that nuclio-platform-config is a configmap that represent nuclio's configuration. once you edit it, you will need to restart both nuclio controller and dashboard deplyoments.

**Comment by IgorKurylo** [comment]
OK, do you have the option to change ingressClassName?

**Comment by liranbg** [comment]
hmm, good question. perhaps @TomerShor will know. is nginx IngressClassName configurable for nuclio functions?

**Comment by TomerShor** [comment]
@liranbg @IgorKurylo nginx IngressClassName is not configurable in nuclio functions.
You can configure an Ingress' host, paths, path type and TLS configuration.

**Comment by liranbg** [comment]
@IgorKurylo  - let us know if there's anything else we can help with.

**Comment by IgorKurylo** [comment]
Thanks, guys.
